File: Classwork/AP_Computer_Science_Ⅰ/src/learnGraphics/pong/Pong.java
package Classwork.AP_Computer_Science_Ⅰ.src.learnGraphics.pong;

import javax.swing.*;
import java.awt.*;

import static javax.swing.WindowConstants.EXIT_ON_CLOSE;

public class Pong extends Thread {
	private static final Dimension dimension = new Dimension(900, 500);
	private final Thread thread = new Thread(this);
	private final JFrame frame;

	private Pong() {
		frame = new JFrame("Pong");
	}

	public static void main(String[] args) {
		new Pong().thread.start();
	}

	@Override
	public void run() {
		frame.setSize(dimension);
		frame.setResizable(false);
		frame.setDefaultCloseOperation(EXIT_ON_CLOSE);
		frame.add(new PongPanel(this));
		frame.setVisible(true);
	}

	public Thread getThread() {
		return thread;
	}

	public JFrame getFrame() {
		return frame;
	}
}

File: Classwork/AP_Computer_Science_Ⅰ/src/learnGraphics/pong/Pong.kt
package learnGraphics.pong

import java.awt.Dimension
import javax.swing.JFrame
import javax.swing.WindowConstants.EXIT_ON_CLOSE

class Pong : Thread() {
	val frame = JFrame("Pong")

	override fun run() {
		frame.size = dimension
		frame.isResizable = false
		frame.defaultCloseOperation = EXIT_ON_CLOSE
		frame.add(PongPanel(this))
		frame.isVisible = true
	}

	companion object {
		private val dimension = Dimension(900, 500)
	}
}
fun main() {
	Pong().start()
}
